EFCore3.Data/UnitOfWork.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EFCore5.Data
+{
+    public class UnitOfWork : IDisposable
+    {
+        private DataContext db;
+        private UserRepository userRepository;
```

Comment:
it should be "IRepository<User>" type
for other repos same logic

and u hould inject them into constructor